src/seaport.ts

```diff
@@ -82,6 +83,7 @@ export class Seaport {
   /**
    * @param provider - The provider to use for web3-related calls
    * @param considerationConfig - A config to provide flexibility in the usage of Seaport
+   * @param signer - The signer used to sign your accounts
    */
   public constructor(
     provider: providers.JsonRpcProvider,
```

Comment:
do people need to provide a `provider` usually if they're going for the custom signer approach?

Wondering if it's better to just make this `providerOrSigner`, and then assign `this.provider` based on the signer's provider if the provided object is a signer otherwise, just the provider.